src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -136,101 +131,39 @@ StageWithCaseData getActiveStage(UUID caseUUID, UUID stageUUID) {
-        if (activeStages.size() > 1) {
-            log.warn("More than 1 active stage for caseUUID: {}, all active stages will be deactivated", caseUUID);
-        }
-
-        Optional<Stage> existingStageOfRequestedType = allStages.stream().filter(stage -> stage.getStageType().equals(createStageRequest.getType())).findFirst();
-
-        final AtomicBoolean isRecreation = new AtomicBoolean(false);
-        final AtomicBoolean isRecreateOfCurrentlyActive = new AtomicBoolean(false);
-        final AtomicReference<Stage> stageToActivate = new AtomicReference<>(null);
-
-        existingStageOfRequestedType.ifPresentOrElse((Stage stage) -> {
-            log.debug("Stage type of {} exists for caseUUID: {}, recreating stage",createStageRequest.getType(), caseUUID);
-            stageToActivate.set(stage);
-            isRecreation.set(true);
-            isRecreateOfCurrentlyActive.set(activeStages.contains(stage));
-        },() -> {
-            log.debug("Stage type of {} does not currently exist for caseUUID: {}, creating new stage",createStageRequest.getType(), caseUUID);
-            stageToActivate.set(new Stage(caseUUID, createStageRequest.getType(),
-                    null, null, createStageRequest.getTransitionNoteUUID()));
-        });
 
-        CaseData caseData = caseDataService.getCase(caseUUID);
-        calculateDeadlines(stageToActivate.get(), caseData);
-
-        stageRepository.save(stageToActivate.get());
-        log.info("Created Stage: {}, Type: {}, Case: {}", stageToActivate.get().getUuid(), stageToActivate.get().getStageType(), stageToActivate.get().getCaseUUID(), value(EVENT, STAGE_CREATED));
-
-        caseDataService.updateCaseData(caseData, stageToActivate.get().getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stageToActivate.get().getStageType()));
-
-        List<UUID> assignedUserUUIDList = activeStages.stream().map(BaseStage::getUserUUID).collect(Collectors.toList());
-        UUID currentUserUUID = null;
-
-        if (!assignedUserUUIDList.isEmpty()) {
-            currentUserUUID = assignedUserUUIDList.get(0);
-        }
-
-        assignTeamAndMemberUserToStage(stageToActivate.get(), createStageRequest.getTeamUUID(), createStageRequest.getUserUUID());
-
-        // deactivate old active stage/s if new stage not current stage
-        if (!isRecreateOfCurrentlyActive.get()) {
-            activeStages.forEach(value -> updateStageTeam(value, caseData.getDataMap(), caseData.getReference(), null, null));
-        }
-
-        // Update audit and timeline unless active and allocated to same user and team - positioned here maintains timeline consistency.
-        if (isRecreation.get() && !isRecreateOfCurrentlyActive.get()) {
-            auditClient.recreateStage(stageToActivate.get());
-        } else if (!isRecreation.get()) {
-            auditClient.createStage(stageToActivate.get());
-        }
+        // Store the current user before clearing it later
+        UUID currentUserUUID = allStages
```

Comment:
This is only used in the assignment notification - if there is more than one stage active then this is flawed and could end up sending an email to the wrong person. 

Would a better solution be adding a check constraint on the stage table to ensure only 1 active stage?

Reply:
it was designed to support parallel stages, but the system has moved too far away in other areas, we're probably better off splitting the table into a stage and current_stage with the team,user and fk back to stage and case